src/main/java/seedu/address/model/tag/TagSet.java

```diff
@@ -0,0 +1,64 @@
+package seedu.address.model.tag;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+
+/**
+ * A set of tags that does not allow nulls.
+ */
+public class TagSet implements Iterable<Tag> {
+    private final Set<Tag> internalSet = new HashSet<>();
+
+    /**
+     * Adds a tag to the set.
+     */
+    public void add(Tag toAdd) {
+        requireNonNull(toAdd);
+        internalSet.add(toAdd);
+    }
+
+    /**
+     * Removes the specified tag from the set.
+     */
```

Comment:
i think you missed out the params attribute for some of the javadocs comments

Reply:
I agree, I think we should include it for consistency, and also by convention.